RATapi/classlist.py

```diff
@@ -296,11 +296,15 @@ def extend(self, other: Sequence[T]) -> None:
         self._check_unique_name_fields(other)
         self.data.extend(other)
 
-    def set_fields(self, index: int, **kwargs) -> None:
+    def set_fields(self, index: Union[int, slice, str, T], **kwargs) -> None:
         """Assign the values of an existing object's attributes using keyword arguments."""
         self._validate_name_field(kwargs)
         pydantic_object = False
 
+        # Find index if name or object is supplied
+        if isinstance(index, (str, self._class_handle)):
+            index = self.index(index)
```

Comment:
Should we make the search case insensitive?

Reply:
It should be case insensitive in `_get_item_from_name_field`